File: from-3101-to-3200/3174-clear-digits/solution.go
func clearDigits(s string) string {
    marks := make([]int, len(s))
    for i := 0; i < len(s); i ++ {
        if s[i] >= 'a' && s[i] <= 'z' {
            marks[i] = 1
        } else {
            for j := i - 1; j >= 0; j -- {
                if marks[j] == 1 {
                    marks[j] = 0
                    break
                }
            }
        }
    }

    res := []string{}
    for i := 0; i < len(s); i ++ {
        if marks[i] == 1 {
            res = append(res, string(s[i]))
        }
    }
    return strings.Join(res, "")
}

File: from-3101-to-3200/3174-clear-digits/solution.py
class Solution:
    def clearDigits(self, s: str) -> str:
        n = len(s)
        marks = [0] * n
        for i, c in enumerate(s):
            if 'a' <= c <= 'z':
                marks[i] = 1
            else:
                for j in range(i - 1, -1, -1):
                    if marks[j] == 1:
                        marks[j] = 0
                        break

        return ''.join(c for c, mark in zip(s, marks) if mark == 1)